src/WebJobs.Script/Binding/ServiceBusScriptBindingProvider.cs

```diff
@@ -110,8 +112,17 @@ public override Type DefaultType
                 {
                     if (Context.Access == FileAccess.Read)
                     {
-                        return string.Compare("binary", Context.DataType, StringComparison.OrdinalIgnoreCase) == 0
+                        Type type = string.Compare("binary", Context.DataType, StringComparison.OrdinalIgnoreCase) == 0
                             ? typeof(byte[]) : typeof(string);
+
+                        if (string.Compare("many", Context.Cardinality, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            // arrays are supported for both trigger input as well
+                            // as output bindings
+                            type = type.MakeArrayType();
+                        }
```

Comment:
There's a code-smell here. Can this be pushed to the caller? 

1. It's eventhub-specific binding information in Script; but we already have captured these semantics in the EventHub WebJobs SDK piece via the rules it supports. 
2. this may need to be repeated for other bindings.  
3. use string comparison for enum checks. 
 
Between knowing Context.DataType & Context.Cardinality; plus the rules exposed by the raw extension, does Script have enough information for us to just eliminate this? And not require each extension author to do write code like this?